Modules/_ssl.c

```diff
@@ -1524,10 +1530,16 @@ _get_crl_dp(X509 *certificate) {
             int err;
 
             gn = sk_GENERAL_NAME_value(gns, j);
+            if (gn == NULL) {
+                goto invalid;
+            }
             if (gn->type != GEN_URI) {
                 continue;
             }
             uri = gn->d.uniformResourceIdentifier;
+            if (uri == NULL) {
```

Comment:
Can you get `uri == NULL` when `type == GEN_URI`?

Reply:
OpenSSL doesn't check both for NULL, so it's not necessary. I checked the code that is eventually called from ``X509_print*()`` API.